src/views/ApartmentApp/components/StudentApplication/index.js

```diff
@@ -26,6 +26,12 @@ const BLANK_APPLICATION_DETAILS = {
   Applicants: [],
   ApartmentChoices: [],
 };
+const INITIAL_DIALOG_PROPS = {
+  labelledby: 'default-dialog',
+  describedby: 'dialog-undefined',
+  title: 'How did you get here?',
+  text: 'This text should not be displayed.',
+};
```

Comment:
I think you could simplify the dialog usage further by defining all the static dialog props here:
```suggestion
const DIALOG_PROPS = {
  default: {
    labelledby: 'default-dialog',
    describedby: 'dialog-undefined',
    title: 'How did you get here?',
    text: 'This text should not be displayed.',
  },
  ...,
  submit: {
    labelledby: 'submit-application-dialog',
    describedby: 'submit-application',
    title: 'Submit apartment application?',
    text: 'Please confirm that all the information you have entered is valid\nClick "Accept" below to submit this application',
  },
};
```

Then, instead of having two separate state variables, you could just call `setDialogProps(DIALOG_PROPS.submit)` for example.

Reply:
Oh nice! That is far more convenient
I was specifically trying to find a better way to handle the static stuff like this, so thanks!

One problem I was having, which inspired the behavior of leaving it unchanged on default, for example, was that if you close accept or close, you would see the default text showup inside the dialog a split-second before the dialog box closed, which just felt a bit sloppy to look at, so I tried to have it leave the props unchanged when it closes the dialog. Though, I guess my comment here is more about the behavior of the switch-case.
Really the problem was that the state updates faster than the DialogBox realizes that its `open` prop changed to false and then it performs the closing animation, so it's a race condition in state change vs. visual animation.

Reply:
Oh yes, I remember encountering the same issue when trying to simplify the API for snackbars. The way I handled it was to only update the `open` boolean to false when the dialog is closed, and to update the content and set `open` to true when a new dialog is opened.

Reply:
Oh right! the snackbars do the `setSnackbar((s) => {...s, open: false})` or something like that, I remember copying that idea from another 360 component somewhere and thinking that it was a very elegant solution 😆